reporter-opentsdb/src/main/java/io/ultrabrew/metrics/reporters/opentsdb/OpenTSDBHttpClient.java

```diff
@@ -77,6 +79,30 @@ void write(final String metricName,
       final String[] tags,
       final long timestamp,
       final String value) throws IOException {
+    // validation
+    if (Strings.isNullOrEmpty(metricName)) {
+      LOG.warn("Null or empty metric name.");
+      return;
+    }
+    if (Strings.isNullOrEmpty(value)) {
+      LOG.warn("Null or empty value.");
+      return;
+    }
+    if (tags == null) {
+      LOG.warn("At least one tag pair must be present.");
+      return;
+    }
+    if (tags.length % 2 != 0) {
+      LOG.warn("Uneven tag count: {} for metric {}", Arrays.toString(tags), metricName);
+      return;
+    }
+    for (int i = 0; i < tags.length; i++) {
+      if (Strings.isNullOrEmpty(tags[i])) {
```

Comment:
In `InfluxDBReporter` `null` tag _values_ are sent as a fixed `NULL` string, should that be the case here also?

Reply:
No, null keys should always throw an exception as it's an invalid situation.

Reply:
But won't this throw for both keys _and_ values?

Reply:
ah whoops sorry this is the TSD reporter. yeah I'll add the null for values too I 'spose.